### ISSUE TITLE
Problem with split function

### ISSUE THREAD

**Issue by Nosenation** [ben-eb/postcss-merge-rules#3]
Hi
I got some very odd problem with you plugin. I try to use it with gulp, but every time I launch it i get error on line 93 of index.js.

![zrzut ekranu 2015-06-15 o 15 41 44](https://cloud.githubusercontent.com/assets/1431712/8161148/49816ab0-1375-11e5-8604-8b71a31f541f.png)

My gulp task looks like this:
```javascript
var
    gulp = require('gulp'),
    postCSS = require('gulp-postcss'), (...)

gulp.task('postCSS', function() {
    return gulp.src(srcAssets.styles + 'sprite.css')
        .pipe(postCSS([require('postcss-merge-rules')()]))
        .pipe(plugins.size({showFiles: true}))
        .pipe(gulp.dest(srcAssets.styles));
});
```

I also tried this version:
```javascript
var
    gulp = require('gulp'),
    postCSS = require('gulp-postcss'),
    mergeRules = require('postcss-merge-rules'), (...)

gulp.task('postCSS', function() {
    return gulp.src(srcAssets.styles + 'sprite.css')
        .pipe(postCSS([mergeRules]))
        .pipe(plugins.size({showFiles: true}))
        .pipe(gulp.dest(srcAssets.styles));
});
```

**Comment by ben-eb** [comment]
Is there some specific CSS that this crashes on?

**Comment by Nosenation** [comment]
I tried it on my old legacy code. I need this to move all bg declarations with same url to one rule :)

**Comment by ben-eb** [comment]
I need to be able to reproduce the error, so if you can find some minimal test case which the module fails on then that would be helpful. :smile:

**Comment by Nosenation** [comment]
I tried it on this file:
```javascript
.btn {
    background: url('../img/sprite_full.png') no-repeat 0 0;
    background-position: -0px -200px;
    border: 0;
    color: #fff;
    font: bold 11px Arial, sans-serif;
    color: #fff;
    cursor: pointer;
    padding: 0 3px;
}

#bg_blk .path_next {
    background: url('../img/sprite_full.png') no-repeat 50% 50%;
    background-position: -328px -195px;
    padding: 0 3px;
}
```
It doesn't show error, but nothing is generated.

**Comment by ben-eb** [comment]
It is outputting for me on node 0.12.2. What version are you using?

**Comment by Nosenation** [comment]
Got the same version and still nothing. Maybe I do something wrong.

My gulpfile:
```javascript
/*******************************************************************************
1. DEPENDENCIES AND SETTINGS
*******************************************************************************/

var
    gulp = require('gulp'),
    plugins = require('gulp-load-plugins')(),
    argv = require('yargs').argv,
    del = require('del'),
    download = require('download'),
    fs = require('fs'),
    lazypipe = require('lazypipe'),
    mkdirp = require('mkdirp'),
    mergeRules = require('postcss-merge-rules'),
    pkg = require('./package.json'),
    runSequence = require('run-sequence'),
    basePaths = {
        dev:  './src/',
        ext: './external/',
        build: './build/'
    },
    destAssets = {
        scripts: basePaths.build + 'js/',
        styles: basePaths.build + 'css/'
    },
    srcAssets = {
        scripts: basePaths.dev + 'js/',
        styles: basePaths.dev + 'css/',
        images: basePaths.dev + 'img/'
    },
    destScripts,
    destStyles,
    versionNumber;

/*******************************************************************************
2. UTILITY TASKS
*******************************************************************************/

var gzipTask = lazypipe()
    .pipe(plugins.gzip, {append: false})
    .pipe(plugins.rename, function(path) {
        path.extname += 'gz';
    })
    .pipe(plugins.size, {showFiles: true});

gulp.task('clean', function() {
    del([basePaths.ext + '*']);
});

// only used in build task, do not run it
gulp.task('version.html', function() {
    versionNumber = pkg.version;
    destScripts = destAssets.scripts.replace('./', '/');
    destStyles = destAssets.styles.replace('./', '/');

    mkdirp(basePaths.build, function() {
        fs.writeFile(basePaths.build + 'prodJS.html', '<script type="text/javascript" src="http://static.im-g.pl/static/forum/front/master' + destScripts + 'main-min.jsgz?v=' + versionNumber + '"></script>');
        fs.writeFile(basePaths.build + 'testJS.html', '<script type="text/javascript" src="http://static.im-g.pl/static/forum/front/develop' + destScripts + 'main-min.jsgz?v=' + versionNumber + '"></script>');
        fs.writeFile(basePaths.build + 'prodCSS.html', '<link rel="stylesheet" href="http://static.im-g.pl/static/forum/front/master' + destStyles + 'style-min.cssgz?v=' + versionNumber + '" type="text/css">');
        fs.writeFile(basePaths.build + 'testCSS.html', '<link rel="stylesheet" href="http://static.im-g.pl/static/forum/front/develop' + destStyles + 'style-min.cssgz?v=' + versionNumber + '" type="text/css">');
    });
});

gulp.task('sprite', function() {
    return gulp.src(srcAssets.styles + 'style.css')
        .pipe(plugins.cssSpriter({
            'spriteSheet': srcAssets.images + 'sprite_full.png',
            'pathToSpriteSheetFromCSS': '../img/sprite_full.png'
        }))
        .pipe(plugins.rename('sprite.css'))
        .pipe(plugins.size({showFiles: true}))
        .pipe(gulp.dest(srcAssets.styles));
});

gulp.task('base64', function() {
    return gulp.src(srcAssets.styles + 'sprite.css')
        .pipe(plugins.cssBase64())
        .pipe(plugins.size({showFiles: true}))
        .pipe(gulp.dest(srcAssets.styles));
});

gulp.task('postCSS', function() {
    return gulp.src(srcAssets.styles + 'test.css')
        .pipe(plugins.postcss([mergeRules]))
        .pipe(plugins.rename('result.css'))
        .pipe(plugins.size({showFiles: true}))
        .pipe(gulp.dest(srcAssets.styles));
});

/*******************************************************************************
3. CREATE CSS AND JS FILES
*******************************************************************************/

gulp.task('css', function() {
    return gulp.src([basePaths.ext + '*.css', srcAssets.styles + 'sprite.css'])
        .pipe(plugins.concat('style-dev.css'))
        .pipe(plugins.size({showFiles: true}))
        .pipe(gulp.dest(destAssets.styles))
        .pipe(plugins.minifyCss())
        .pipe(plugins.rename('style-min.css'))
        .pipe(plugins.size({showFiles: true}))
        .pipe(gulp.dest(destAssets.styles))
        .pipe(gzipTask())
        .pipe(gulp.dest(destAssets.styles));
});

gulp.task('js', function() {
    return gulp.src([basePaths.ext + '*.js', srcAssets.scripts + '*.js'])
        .pipe(plugins.concat('main-dev.js'))
        .pipe(plugins.size({showFiles: true}))
        .pipe(gulp.dest(destAssets.scripts))
        .pipe(plugins.uglify())
        .pipe(plugins.rename('main-min.js'))
        .pipe(plugins.size({showFiles: true}))
        .pipe(gulp.dest(destAssets.scripts))
        .pipe(gzipTask())
        .pipe(gulp.dest(destAssets.scripts));
});

/*******************************************************************************
4. BUILD TASKS
*******************************************************************************/

gulp.task('build', ['clean'], function() {
    new download({mode: '755'})
        .get('http://bi.im-g.pl/info/ban.css')
        .get('http://bi.im-g.pl/css/globalshared.css')
        .get('http://bi.im-g.pl/info/cookie_profil.js')
        .get('http://bi.im-g.pl/info/util.js')
        .get('http://bi.im-g.pl/info/forum/bbcodeSonda.js')
        .get('http://bi.im-g.pl/info/jquery-1.4.4.js')
        .dest(basePaths.ext)
        .run(function() {
            runSequence(['css', 'js'], 'version.html');
        });
});
```

**Comment by ben-eb** [comment]
@ai Do you have any ideas?

**Comment by Nosenation** [comment]
Small update.

Checked on this file:
```css
.btn {
    padding: 0 3px;
    background: url('../img/sprite_full.png') no-repeat;
    background-position: -0px -200px;
    border: 0;
    color: #fff;
    font: bold 11px Arial, sans-serif;
    color: #fff;
    cursor: pointer;
}

.bg_blk .path_next {
    padding: 0 3px;
    background: url('../img/sprite_full.png') no-repeat;
    background-position: -328px -195px;
}

h1 {
    color: blue;
}

h2 {
    color: blue;
    font-size: 12px;
}
```

and it only worked for non-class, non-id selectors. Result:
```css
.btn {
    padding: 0 3px;
    background: url('../img/sprite_full.png') no-repeat;
    background-position: -0px -200px;
    border: 0;
    color: #fff;
    font: bold 11px Arial, sans-serif;
    color: #fff;
    cursor: pointer;
}

.bg_blk .path_next {
    padding: 0 3px;
    background: url('../img/sprite_full.png') no-repeat;
    background-position: -328px -195px;
}

h1,h2 {
    color: blue;
}

h2 {
    font-size: 12px;
}
```

So it works, kind off. I want to get this type of result:
```css
.btn, .bg_blk .path_next {
    padding: 0 3px;
    background: url('../img/sprite_full.png') no-repeat;
}

.btn {
    background-position: -0px -200px;
    border: 0;
    color: #fff;
    font: bold 11px Arial, sans-serif;
    color: #fff;
    cursor: pointer;
}

.bg_blk .path_next {
    background-position: -328px -195px;
}

h1,h2 {
    color: blue;
}

h2 {
    font-size: 12px;
}
```

**Comment by ben-eb** [comment]
This might be an issue with the logic for (not) merging shorthand properties.

**Comment by ben-eb** [comment]
Fixed the crashing behaviour in `1.3.2`. The rest might be a duplicate of https://github.com/ben-eb/cssnano/issues/36.
